pkg/kn/commands/source/apiserver/apiserver_create.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiserver
+
+import (
+	"errors"
+
+	"github.com/spf13/cobra"
+
+	"knative.dev/client/pkg/kn/commands"
+	"knative.dev/client/pkg/kn/commands/source/sink"
+)
+
+func NewApiServerCreateCommand(p *commands.KnParams) *cobra.Command {
+	var apiServerUpdateFlags ApiServerSourceUpdateFlags
+	var sinkFlags sink.SinkFlags
+
+	cmd := &cobra.Command{
+		Use:   "create NAME --resources RESOURCE --accountname ACCOUNTNAME --sink SINK --mode MODE",
+		Short: "Create a Kubernetes Api Server event source",
+		Example: `
+  # Create a Kubernetes Api Server event source 'k8sevents' which consumes Kubernetes events and send them to service 'mysvc' as a cloudevent
+  kn source api-server create k8sevents --resources Event --accountname myaccountname --sink svc:mysvc`,
+
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			if len(args) != 1 {
+				return errors.New("'source api-server create' requires the name of the source as single argument")
+			}
+			name := args[0]
+
+			namespace, err := p.GetNamespace(cmd)
+			if err != nil {
+				return err
+			}
+
+			servingClient, err := p.NewClient(namespace)
+			if err != nil {
+				return err
+			}
+
+			config, err := p.GetClientConfig()
+			if err != nil {
+				return err
+			}
+			aipServerSourceClient, err := NewEventingClient(config, namespace)
+
+			objectRef, err := sinkFlags.ResolveSink(servingClient)
+			if err != nil {
+				return err
+			}
+			return aipServerSourceClient.CreateApiServerSource(name, apiServerUpdateFlags.getApiServerResourceArray(), apiServerUpdateFlags.accountName, apiServerUpdateFlags.mode, objectRef)
```

Comment:
This reads odd and could be improve a bit by unpacking the return. For instance:

```go
resourceArray := apiServerUpdateFlags.getApiServerResourceArray()
serverSource := aipServerSourceClient.CreateApiServerSource(name, resourceArray, apiServerUpdateFlags.accountName, apiServerUpdateFlags.mode, objectRef)
return serverSource
```

Also can any of these `CreateApiXxx(...)` call produce errors?

Reply:
CreateApiServerSource could produce error. In the current code base, it has been changed to `return aipServerSourceClient.CreateApiServerSource(...)`. So the error will be returned if it is not nil.